api/src/main/java/org/openmrs/Location.java

```diff
@@ -131,6 +133,22 @@ public void setAddress2(String address2) {
 		this.address2 = address2;
 	}
 	
+	/**
+	 * @return the startDate
+	 * @since 1.9
+	 */
+	public Date getStartDate() {
+		return startDate;
+	}
+	
+	/**
+	 * @param startDate to set to
+	 * @since 1.9
+	 */
+	public void setStartDate(Date startDate) {
+		this.startDate = startDate;
+	}
```

Comment:
Hii @kdaud, can you have a look at [this](http://www.informit.com/articles/article.aspx?p=31551&seqNum=2), when working with mutable objects, it's recommended to make deep copies to prevent memory leaks :)

Reply:
@PrathameshMutkure This is a setter method. Its not for cloning the object

Reply:
Well, that's where the problem lies, can you go through [this](https://stackoverflow.com/questions/18954873/malicious-code-vulnerability-may-expose-internal-representation-by-incorporati/18954919#18954919), Date being mutable can be changed unintentionally by external source which reflects in the value of actual object. In our case I'm talking about the `startDate` object which should be deep copied inside getters and setters

Hope you get my point :)

Reply:
@PrathameshMutkure Yeah i get your point. But i believe our code should reflect the changes done by the user. This method is called setDate and it should just set the given date to the object. If the user change the date after setting it, then it should reflect the change to the object as well.

Reply:
Hi, i completely agree with @dilanthas because i thought that if user commited a change then he/she should be able to review the change made by him/her and this is done by setData .